Content.Server/BluespaceHarvester/BluespaceHarvesterComponent.cs

```diff
@@ -0,0 +1,103 @@
+    public int CurrentLevel = 0;
+
+    [DataField("targetLevel"), ViewVariables(VVAccess.ReadWrite)]
+    public int TargetLevel = 0;
+
+    [DataField("maxLevel")]
+    public int MaxLevel = 20;
+
+    [DataField("redspaceTap")]
+    public BluespaceHarvesterVisuals RedspaceTap = BluespaceHarvesterVisuals.TapRedspace;
+
+    [DataField("stabelLevel")]
+    public int StableLevel = 10;
+
+    [DataField("emaggedStableLevel")]
+    public int EmaggedStableLevel = 5;
+
+    [DataField("points"), ViewVariables(VVAccess.ReadWrite)]
+    public int Points = 0;
+
+    [DataField("totalPoints"), ViewVariables(VVAccess.ReadWrite)]
+    public int TotalPoints = 0;
+
+    [DataField("dangerPoints"), ViewVariables(VVAccess.ReadWrite)]
+    public int Danger = 0;
+
+    [DataField("reseted")]
+    public bool Enable = false;
+
+    [DataField("spawnRadius")]
+    public float SpawnRadius = 5f;
+
+    [DataField("rift", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    public string Rift = "BluespaceHarvesterRift";
+
+    [DataField("riftChance")]
+    public float RiftChance = 0.1f;
+
+    [DataField("emaggedRiftChance")]
+    public float EmaggedRiftChance = 0.05f;
+
+    /// <summary>
+    ///     After this danger value, the generation of dangerous creatures and anomalies will begin.
+    /// </summary>
+    [DataField("dangerLimit")]
+    public int DangerLimit = 175;
+
+    [DataField("categories")]
+    public List<BluespaceHarvesterCategoryInfo> Categories = new()
+    {
+        new BluespaceHarvesterCategoryInfo()
+        {
+            PrototypeId = "RandomHarvesterBiologicalLoot",
+            Cost = 7_500,
+            Type = BluespaceHarvesterCategory.Biological,
+        },
+        new BluespaceHarvesterCategoryInfo()
+        {
+            PrototypeId = "RandomHarvesterTechnologicalLoot",
+            Cost = 10_000,
+            Type = BluespaceHarvesterCategory.Technological,
+        },
+        new BluespaceHarvesterCategoryInfo()
+        {
+            PrototypeId = "RandomHarvesterIndustrialLoot",
+            Cost = 12_500,
+            Type = BluespaceHarvesterCategory.Industrial,
+        },
+        new BluespaceHarvesterCategoryInfo()
+        {
+            PrototypeId = "RandomHarvesterDestructionLoot",
+            Cost = 15_000,
+            Type = BluespaceHarvesterCategory.Destruction,
+        },
+    };
+
+    [DataField("spawnEffect"), ViewVariables(VVAccess.ReadWrite)]
+    public string SpawnEffectPrototype = "EffectEmpPulse";
+
+    [DataField("spawnSound"), ViewVariables(VVAccess.ReadWrite)]
+    public SoundSpecifier SpawnSound = new SoundPathSpecifier("/Audio/Effects/teleport_arrival.ogg");
+}
+
+[Serializable]
+public sealed partial class BluespaceHarvesterTap
+{
+    [DataField("level")]
+    public int Level;
+
+    [DataField("visual")]
+    public BluespaceHarvesterVisuals Visual;
+}
```

Comment:
Data fields don't need to have their name specified explicitly anymore if they are the same as the field in camelCase format.
The ones that I have left out should be fixed through your code editor instead.

Also any data fields using a custom type serializer of type `PrototypeIdSerializer<EntityPrototype>` can have that removed, with the field changed to be of `EntProtoId` type instead of `string` type.
As an example:
This:
```csharp
    [DataField(customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>)), ViewVariables(VVAccess.ReadWrite)]
    public string Rift = "BluespaceHarvesterRift";
```
Becomes:
```csharp
    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public EntProtoId Rift = "BluespaceHarvesterRift";
```
For fields of other prototype types there is ProtoId<T> instead.


```suggestion
    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public int CurrentLevel = 0;

    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public int TargetLevel = 0;

    [DataField]
    public int MaxLevel = 20;

    [DataField]
    public BluespaceHarvesterVisuals RedspaceTap = BluespaceHarvesterVisuals.TapRedspace;

    [DataField("stabelLevel")]
    public int StableLevel = 10;

    [DataField]
    public int EmaggedStableLevel = 5;

    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public int Points = 0;

    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public int TotalPoints = 0;

    [DataField("dangerPoints"), ViewVariables(VVAccess.ReadWrite)]
    public int Danger = 0;

    [DataField("reseted")]
    public bool Enable = false;

    [DataField]
    public float SpawnRadius = 5f;

    [DataField(customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>)), ViewVariables(VVAccess.ReadWrite)]
    public string Rift = "BluespaceHarvesterRift";

    [DataField]
    public float RiftChance = 0.1f;

    [DataField]
    public float EmaggedRiftChance = 0.05f;

    /// <summary>
    ///     After this danger value, the generation of dangerous creatures and anomalies will begin.
    /// </summary>
    [DataField]
    public int DangerLimit = 175;

    [DataField]
    public List<BluespaceHarvesterCategoryInfo> Categories = new()
    {
        new BluespaceHarvesterCategoryInfo()
        {
            PrototypeId = "RandomHarvesterBiologicalLoot",
            Cost = 7_500,
            Type = BluespaceHarvesterCategory.Biological,
        },
        new BluespaceHarvesterCategoryInfo()
        {
            PrototypeId = "RandomHarvesterTechnologicalLoot",
            Cost = 10_000,
            Type = BluespaceHarvesterCategory.Technological,
        },
        new BluespaceHarvesterCategoryInfo()
        {
            PrototypeId = "RandomHarvesterIndustrialLoot",
            Cost = 12_500,
            Type = BluespaceHarvesterCategory.Industrial,
        },
        new BluespaceHarvesterCategoryInfo()
        {
            PrototypeId = "RandomHarvesterDestructionLoot",
            Cost = 15_000,
            Type = BluespaceHarvesterCategory.Destruction,
        },
    };

    [DataField("spawnEffect"), ViewVariables(VVAccess.ReadWrite)]
    public string SpawnEffectPrototype = "EffectEmpPulse";

    [DataField, ViewVariables(VVAccess.ReadWrite)]
    public SoundSpecifier SpawnSound = new SoundPathSpecifier("/Audio/Effects/teleport_arrival.ogg");
}

[Serializable]
public sealed partial class BluespaceHarvesterTap
{
    [DataField]
    public int Level;

    [DataField]
    public BluespaceHarvesterVisuals Visual;
}
```